### DISCUSSION TITLE
Simple Digital Read Input Pin Configuration ESP12/ESP8266 (Tuya Brain-Transplant)

### DISCUSSION THREAD

**Discussion by belveder79** [arendst/Tasmota#20406] [Q&A]
I recently got some Water Leak Detectors from Tuya and - as with all those devices - I made a brain transplant to a ESP8266.

![IMG_8197](https://github.com/arendst/Tasmota/assets/112880/92f2415f-0496-463a-b2be-164a78110616)

It's not exactly this device, but you can get the idea:

https://www.elektroda.com/rtvforum/topic3954826.html

Now I have 3 questions. 

1) Those devices have some voltage divider to provide approximate battery level readings to the ADC pin. What's the correct way to configure the PIN in the configuration? I now have simple ADC reading which gives me something, but I don't know if it makes sense or not.

2) MOST IMPORTANT. It seems that the water level is read on GPIO 4 as a digital signal, but I could not find any way to configure the PIN to be a simple digital input pin (similar to digitalRead on ESP32). How do I do this?

3) I suspect the correct way to use those battery-powered devices is to use the deep-sleep functionality with a timer to wake up and go to sleep with rules, right?

Any help highly appreciated.
Thx
Clemens

**Comment by sfromis** [comment]
1. You can config the analog input (gpio 17) for `ADC Range` and then use the command [AdcParam](https://tasmota.github.io/docs/ADC/#commands), option 6 for linear range remapping.
2. The usual way for a digital input is to config as a [Switch](https://tasmota.github.io/docs/Buttons-and-Switches/#switch), and select a SwitchMode to match the behavior. Then you can have rules to trigger when something happens. You can also have rules to react to the state when booting. In other words, as rules are based on events, you do not make a "read", but instead let Tasmota trigger a rule when it has a value for you.
3. Yes, that's [reasonable](https://tasmota.github.io/docs/DeepSleep/).

**Reply by belveder79** [reply]
Thx.... I think I need to come up with a diagram of the board before I move on, as it does not seem to do anything reasonable actually... I also can't make up any change on the gpios whether the water detector is shorted or not...

**Reply by sfromis** [reply]
A simple test case for detecting "signs of life" is to config an input as a counter. Then Tasmota will increment the displayed count value each "cycle".
